### PR TITLE
docs: add a detailed Next.js integration guide

### PR DESCRIPTION
**Pull request by hanspagel**
I was testing the Next.js integration today and used the opportunity to write a detailed integration guide.

@amritk I’m not really sure I did a good job describing what to use when, would you mind to check it’s not completely off?

### CODE REVIEW

**Review by marclave — APPROVED** [review]

**Review by amritk — APPROVED** [review]
😅 haha didn't need to explain how to install next, but doesn't hurt! This will change once we build an integration anyway.

**Review by amritk — CHANGES_REQUESTED** [review]
Actually can we specify app router for the top example, then page router for the bottom? not quite clear

File: documentation/nextjs.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+## Render your OpenAPI reference with Scalar
+
+Ready to add your API reference? Cool, there are a few options to integrate your API reference. The recommended way is to use our Next.js integration for app routing:
+
+```bash
+npm add @scalar/nextjs-api-reference
+```
+
+… and add a new app route:
```
Comment (amritk) [inline documentation/nextjs.md:41]:
```suggestion
### App router

… and add a new app route:
```

File: documentation/nextjs.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  spec: {
+    content: 'https://cdn.jsdelivr.net/npm/@scalar/galaxy/dist/latest.yaml',
+  },
+}
+
+export const GET = ApiReference(config)
+```
+
+Open <http://localhost:3000/reference> and there it is: Your new API reference. :)
+
```
Comment (amritk) [inline documentation/nextjs.md:57]:
```suggestion

### Page router

```

**Pull request merged**